documentation/modules/upgrading/con-upgrade-cluster.adoc

```diff
@@ -0,0 +1,40 @@
+// This module is included in the following assemblies:
+//
+// upgrading/assembly-upgrade.adoc
+
+[id='con-upgrade-cluster-{context}']
+= Upgrading Kubernetes
```

Comment:
I think this is a bit weird:
* The first part focus on how to upgrade Kubernetes cluster. It seems to be a strange mixture of specific and unspecific steps which risk that you forget something and does not really help in any case. I think the whole part should be covered by this single sentence:
  ```
  Refer to your Kubernetes distribution upgrade documentation to check the upgrade path and the steps to upgrade your nodes correctly.
  ```
  I would not leave anything else.
* The second part seems focus on OLM upgrades. But these are completely unrelated to Kubernetes upgrades. Strimzi supports wide variety of Kube versions, so the probability that Kubernetes upgrade enables some new Strimzi upgrade is highly unlikely. In fact it happened only once so far - in a specific constellation when you upgraded to Kube 1.16 after Strimzi 0.22 was released. So I do not think this is useful either.

----

Instead we IMHO need to focus on how to keep the existing clusters fully available during the upgrade. I would expect that almost any upgrade procedure would include shutting down nodes => i.e. first draining them. So the focus here should be on:
* Configuring Pod Disruption Budgets
* Manually rolling the pods while making sure the topics are in-sync / using the Drain Cleaner and operator to do this automatically

Reply:
Yes. I made a comment about this pod disruption and draining when I pushed up the PR. Thanks for confirming.